boards/arm/frdm_k64f/doc/index.rst

```diff
@@ -61,43 +61,89 @@ Supported Features
-| PINMUX    | on-chip    | pinmux                              |
-+-----------+------------+-------------------------------------+
-| GPIO      | on-chip    | gpio                                |
-+-----------+------------+-------------------------------------+
-| I2C       | on-chip    | i2c                                 |
-+-----------+------------+-------------------------------------+
-| SPI       | on-chip    | spi                                 |
-+-----------+------------+-------------------------------------+
-| WATCHDOG  | on-chip    | watchdog                            |
-+-----------+------------+-------------------------------------+
-| ADC       | on-chip    | adc                                 |
-+-----------+------------+-------------------------------------+
-| PWM       | on-chip    | pwm                                 |
-+-----------+------------+-------------------------------------+
-| ETHERNET  | on-chip    | ethernet                            |
-+-----------+------------+-------------------------------------+
-| UART      | on-chip    | serial port-polling;                |
-|           |            | serial port-interrupt               |
-+-----------+------------+-------------------------------------+
-| FLASH     | on-chip    | soc flash                           |
-+-----------+------------+-------------------------------------+
-| USB       | on-chip    | USB device                          |
-+-----------+------------+-------------------------------------+
-| SENSOR    | off-chip   | fxos8700 polling;                   |
-|           |            | fxos8700 trigger                    |
-+-----------+------------+-------------------------------------+
-| CAN       | on-chip    | can                                 |
-+-----------+------------+-------------------------------------+
-| RTC       | on-chip    | rtc                                 |
-+-----------+------------+-------------------------------------+
+.. list-table::
+   :header-rows: 1
+
+   * - Interface
+     - Controller
+     - Driver/Component
+
+   * - :ref:`adc_api`
+     - on-chip
+     - :option:`CONFIG_ADC_MCUX_ADC16`
+
+   * - :ref:`can_api`
+     - on-chip
+     - :option:`CONFIG_CAN_MCUX_FLEXCAN`
+
+   * - :ref:`clock_control_api`
+     - on-chip
+     - :option:`CONFIG_CLOCK_CONTROL_MCUX_SIM`
+       :option:`CONFIG_CLOCK_CONTROL_MCUX_MCG`
+
+   * - :ref:`counter_api`
+     - on-chip
+     - :option:`CONFIG_COUNTER_MCUX_RTC`
+
+   * - :ref:`dac_api`
+     - on-chip
+     - :option:`CONFIG_DAC_MCUX_DAC`
+
+   * - :ref:`entropy_api`
+     - on-chip
+     - :option:`CONFIG_ENTROPY_MCUX_RNGA`
+
+   * - :ref:`ethernet_interface`
+     - on-chip
+     - :option:`CONFIG_ETH_MCUX`
+
+   * - :ref:`flash_api`
+     - on-chip
+     - :option:`CONFIG_SOC_FLASH_MCUX`
+
+   * - :ref:`gpio_api`
+     - on-chip
+     - :option:`CONFIG_GPIO_MCUX`
+
+   * - :ref:`i2c_api`
+     - on-chip
+     - :option:`CONFIG_I2C_MCUX`
```

Comment:
I wonder if the option mentioned should be the MCUX dedicated flag or the Zephyr high level symbol `CONFIG_I2C`.
From the application point of view I think we should encourage use of Zephyr flags (which should end up enabling  CONFIG_I2C_MCUX`), and keep (or make) the vendor specific flags hidden from users perspective (no prompt).